### PR TITLE
Add index for assets to ElasticSearch

### PR DESCRIPTION
**Pull request by rstorey**

### CODE REVIEW

File: concordia/documents.py [hunk 1]
```diff
@@ -15,18 +15,21 @@
 site_report = Index("site_reports")
 site_report.settings(number_of_shards=1, number_of_replicas=0)
 
+asset = Index("assets")
+asset.settings(number_of_shards=1, number_of_replicas=0)
```
Comment (acdha) [inline concordia/documents.py]: I don't think this is the time for it but we should probably review the number of shards

File: concordia/documents.py [hunk 2]
```diff
@@ -15,18 +15,21 @@
...
+
 
 @user.doc_type
 class UserDocument(DocType):
     class Meta:
         model = User
 
-        fields = ["last_login", "date_joined", "username"]
+        fields = ["last_login", "date_joined", "username", "is_active"]
```
Comment (acdha) [inline concordia/documents.py]: Do we want to filter on this or restrict the set of users to `is_active=True`? Also, would we want to filter `is_staff=False`?

File: concordia/documents.py [hunk 3]
```diff
@@ -143,3 +146,44 @@ def get_queryset(self):
                 "asset__item", "asset__item__project", "asset__item__project__campaign"
             )
         )
+
+
+@asset.doc_type
+class AssetDocument(DocType):
+    item = fields.ObjectField(
+        properties={
```
Comment (acdha) [inline concordia/documents.py]: Should we include the item creation/modification time?

File: concordia/models.py [hunk 4]
```diff
@@ -234,6 +234,9 @@ def get_absolute_url(self):
             },
         )
 
+    def latest_transcription(self):
+        return self.transcription_set.order_by("-pk").first()
```
Comment (acdha) [inline concordia/models.py]: I'm not opposed to having this as a class method but I'm thinking that for search we might want to have a prefetch to do this in bulk — something like `prefetch_related(latest_transcriptions=Prefetch('transcription_set', queryset=Transcription.objects.order_by('pk').latest())` & an accessor but I would want to test that.

File: concordia/documents.py (hunk 2 above)
Comment (rstorey) [reply concordia/documents.py]: We need to be able to tell how many users haven't activated their accounts, so we want everybody. We also want to be able to tell if any staff are in the top contributors, so we include them too.

File: concordia/documents.py (hunk 3 above)
Comment (rstorey) [reply concordia/documents.py]: We don't have `created_on` or `updated_on` for items either.

File: concordia/models.py (hunk 4 above)
Comment (rstorey) [reply concordia/models.py]: I like the idea of having it pre-fetch. I'm following the documentation for django-elasticsearch-dsl about "Using Different Attributes for Fields" https://pypi.org/project/django-elasticsearch-dsl/ 
If you have an alternative method to propose, I'm interested.

Comment (acdha) [reply concordia/models.py]: Have you measured performance on this currently? I'm assuming it'll be slow enough to be a concern but we should probably confirm that before spending more time on it.

Comment (rstorey) [reply concordia/models.py]: No, I used it once a week ago, since then haven't looked at it.

Comment (rstorey) [reply concordia/models.py]: It's not super slow. I was able to fully rebuild all the indices in less than 15 minutes.

Comment (acdha) [reply concordia/models.py]: Okay, let's table this until we have a problem worth attention

**Pull request merged**